src/geom/SkyWcs.cc

```diff
@@ -0,0 +1,249 @@
+/*
+ * LSST Data Management System
+ * See COPYRIGHT file at the top of the source tree.
+ *
+ * This product includes software developed by the
+ * LSST Project (http://www.lsst.org/).
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the LSST License Statement and
+ * the GNU General Public License along with this program.  If not,
+ * see <http://www.lsstcorp.org/LegalNotices/>.
+ */
+
+#include <exception>
+#include <memory>
+#include <ostream>
+#include <sstream>
+#include <vector>
+
+#include "astshim.h"
+
+#include "lsst/afw/formatters/Utils.h"
+#include "lsst/afw/geom/detail/frameSetUtils.h"
+#include "lsst/afw/geom/Point.h"
+#include "lsst/afw/geom/SpherePoint.h"
+#include "lsst/afw/geom/SkyWcs.h"
+#include "lsst/daf/base/PropertyList.h"
+#include "lsst/pex/exceptions.h"
+
+namespace lsst {
+namespace afw {
+namespace geom {
+namespace {
+
+inline double square(double x) { return x * x; }
```

Comment:
`std::pow(x, 2)` translates to a single multiply in gcc 6.3 and clang 4.0: https://godbolt.org/g/6qLAky